includes/embeds/class-amp-facebook-embed.php

```diff
@@ -199,10 +199,6 @@ private function create_amp_facebook_and_replace_node( $dom, $node, $embed_type
 			'width'  => $node->hasAttribute( 'data-width' ) ? $node->getAttribute( 'data-width' ) : $this->DEFAULT_WIDTH,
 			'height' => $node->hasAttribute( 'data-height' ) ? $node->getAttribute( 'data-height' ) : $this->DEFAULT_HEIGHT,
 		];
-		if ( '100%' === $attributes['width'] ) {
-			$attributes['layout'] = 'fixed-height';
-			$attributes['width']  = 'auto';
-		}
```

Comment:
Where is this conversion of `100%` width to fixed-height `auto` being done now? This was added specifically to address an issue in https://github.com/ampproject/amp-wp/pull/2776.

The code here actually should be generalized to apply to all sanitizers. Anywhere that `width=100%` is found it should be converted into `layout=fixed-height` and `width=auto`.

Reply:
I was under the impression from #2146 that this was a temporary fix, but the sanitizers should remove such cases. Will re-include it.

Reply:
This is now taken care of by `AMP_Tag_And_Attribute_Sanitizer::adapt_dimensions_to_layout()`.